### PR TITLE
TST: jest27, detox to dev deps, remove jasmine

### PR DESCRIPTION
**Pull request by limpbrains**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/qehayj85bbke7v6mmwp2brrmj8?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9870)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/u79pcfnyfb81mwgfxxdpn2b5zr?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9871)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/6t7p1xx5zvcgr1c7h17hc36yfm?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9872)

**Comment by limpbrains** [comment]
`class/payjoin-transaction.js` is leaking somewhere. Jest can't run tests, even with `jest.useFakeTimers()`

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/zg8fh1jdfuab4kn3j7fdcv1t28?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9873)

**Comment by Overtorment** [comment]
detox is not in dev deps because we build a release apk for e2e

**Comment by Overtorment** [comment]
iirc payjoin class does have a timer. its for a case when server didn complete doin a pj, so we broadcast original (non-payjoined) txhex

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/batcfhq4zp0dauxfc1kp3dkg5c?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9874)

**Comment by limpbrains** [comment]
@Overtorment Can't find it and I don't think this is the issue.
Look what `npx jest tests/unit/payjoin-transaction.test.js --detectOpenHandles` shows
I'm confused 

```
  ●  PBKDF2REQUEST

      75 |     const mnemonic = this.secret;
      76 |     const passphrase = this.passphrase;
    > 77 |     return bip39.mnemonicToSeedSync(mnemonic, passphrase);
         |                  ^
      78 |   }
      79 |
      80 |   setSecret(newSecret: string): this {

      at Object.nativePBKDF2Sync [as pbkdf2Sync] (node_modules/pbkdf2/index.js:26:17)
      at Object.mnemonicToSeedSync (node_modules/bip39/src/index.js:55:21)
      at HDSegwitBech32Wallet._getSeed (class/wallets/abstract-hd-wallet.ts:77:18)
      at HDSegwitBech32Wallet.getXpub (class/wallets/abstract-hd-electrum-wallet.js:184:23)
      at HDSegwitBech32Wallet._getNodeAddressByIndex (class/wallets/abstract-hd-electrum-wallet.js:118:55)
      at HDSegwitBech32Wallet._getInternalAddressByIndex (class/wallets/abstract-hd-electrum-wallet.js:170:19)
      at _callee3$ (tests/unit/payjoin-transaction.test.js:34:15)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
      at node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7
      at tryCallTwo (node_modules/react-native/node_modules/promise/lib/core.js:45:5)
      at doResolve (node_modules/react-native/node_modules/promise/lib/core.js:200:13)
      at new Promise (node_modules/react-native/node_modules/promise/lib/core.js:66:3)
      at Object.<anonymous> (node_modules/@babel/runtime/helpers/asyncToGenerator.js:21:12)
      at TestScheduler.scheduleTests (node_modules/@jest/core/build/TestScheduler.js:333:13)
      at runJest (node_modules/@jest/core/build/runJest.js:404:19)
      at _run10000 (node_modules/@jest/core/build/cli/index.js:320:7)
      at runCLI (node_modules/@jest/core/build/cli/index.js:173:3)


  ●  PBKDF2REQUEST

      75 |     const mnemonic = this.secret;
      76 |     const passphrase = this.passphrase;
    > 77 |     return bip39.mnemonicToSeedSync(mnemonic, passphrase);
         |                  ^
      78 |   }
      79 |
      80 |   setSecret(newSecret: string): this {

      at Object.nativePBKDF2Sync [as pbkdf2Sync] (node_modules/pbkdf2/index.js:26:17)
      at Object.mnemonicToSeedSync (node_modules/bip39/src/index.js:55:21)
      at HDSegwitBech32Wallet._getSeed (class/wallets/abstract-hd-wallet.ts:77:18)
      at HDSegwitBech32Wallet._getWIFByIndex (class/wallets/abstract-hd-electrum-wallet.js:93:23)
      at HDSegwitBech32Wallet._getExternalWIFByIndex (class/wallets/abstract-hd-electrum-wallet.js:77:19)
      at HDSegwitBech32Wallet._getWifForAddress (class/wallets/abstract-hd-wallet.ts:273:60)
      at class/wallets/abstract-hd-electrum-wallet.js:890:43
          at Array.forEach (<anonymous>)
      at HDSegwitBech32Wallet.createTransaction (class/wallets/abstract-hd-electrum-wallet.js:886:14)
      at _callee3$ (tests/unit/payjoin-transaction.test.js:30:15)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
      at node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7
      at tryCallTwo (node_modules/react-native/node_modules/promise/lib/core.js:45:5)
      at doResolve (node_modules/react-native/node_modules/promise/lib/core.js:200:13)
      at new Promise (node_modules/react-native/node_modules/promise/lib/core.js:66:3)
      at Object.<anonymous> (node_modules/@babel/runtime/helpers/asyncToGenerator.js:21:12)
      at TestScheduler.scheduleTests (node_modules/@jest/core/build/TestScheduler.js:333:13)
      at runJest (node_modules/@jest/core/build/runJest.js:404:19)
      at _run10000 (node_modules/@jest/core/build/cli/index.js:320:7)
      at runCLI (node_modules/@jest/core/build/cli/index.js:173:3)


  ●  PBKDF2REQUEST

      75 |     const mnemonic = this.secret;
      76 |     const passphrase = this.passphrase;
    > 77 |     return bip39.mnemonicToSeedSync(mnemonic, passphrase);
         |                  ^
      78 |   }
      79 |
      80 |   setSecret(newSecret: string): this {

      at Object.nativePBKDF2Sync [as pbkdf2Sync] (node_modules/pbkdf2/index.js:26:17)
      at Object.mnemonicToSeedSync (node_modules/bip39/src/index.js:55:21)
      at HDSegwitBech32Wallet._getSeed (class/wallets/abstract-hd-wallet.ts:77:18)
      at HDSegwitBech32Wallet.getXpub (class/wallets/abstract-hd-electrum-wallet.js:184:23)
      at HDSegwitBech32Wallet._getNodeAddressByIndex (class/wallets/abstract-hd-electrum-wallet.js:118:55)
      at HDSegwitBech32Wallet._getInternalAddressByIndex (class/wallets/abstract-hd-electrum-wallet.js:170:19)
      at HDSegwitBech32Wallet._getWifForAddress (class/wallets/abstract-hd-wallet.ts:264:36)
      at PayjoinTransaction._callee$ (class/payjoin-transaction.js:27:38)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
      at node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7
      at tryCallTwo (node_modules/react-native/node_modules/promise/lib/core.js:45:5)
      at doResolve (node_modules/react-native/node_modules/promise/lib/core.js:200:13)
      at new Promise (node_modules/react-native/node_modules/promise/lib/core.js:66:3)
      at PayjoinTransaction.<anonymous> (node_modules/@babel/runtime/helpers/asyncToGenerator.js:21:12)
      at PayjoinTransaction.getPsbt (class/payjoin-transaction.js:33:131)
      at PayjoinClient.run (node_modules/payjoin-client/src/client.js:37:36)
      at _callee6$ (tests/unit/payjoin-transaction.test.js:123:29)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
      at node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7
      at tryCallTwo (node_modules/react-native/node_modules/promise/lib/core.js:45:5)
      at doResolve (node_modules/react-native/node_modules/promise/lib/core.js:200:13)
      at new Promise (node_modules/react-native/node_modules/promise/lib/core.js:66:3)
      at Object.<anonymous> (node_modules/@babel/runtime/helpers/asyncToGenerator.js:21:12)
      at TestScheduler.scheduleTests (node_modules/@jest/core/build/TestScheduler.js:333:13)
      at runJest (node_modules/@jest/core/build/runJest.js:404:19)
      at _run10000 (node_modules/@jest/core/build/cli/index.js:320:7)
      at runCLI (node_modules/@jest/core/build/cli/index.js:173:3)


  ●  PBKDF2REQUEST

      75 |     const mnemonic = this.secret;
      76 |     const passphrase = this.passphrase;
    > 77 |     return bip39.mnemonicToSeedSync(mnemonic, passphrase);
         |                  ^
      78 |   }
      79 |
      80 |   setSecret(newSecret: string): this {

      at Object.nativePBKDF2Sync [as pbkdf2Sync] (node_modules/pbkdf2/index.js:26:17)
      at Object.mnemonicToSeedSync (node_modules/bip39/src/index.js:55:21)
      at HDSegwitBech32Wallet._getSeed (class/wallets/abstract-hd-wallet.ts:77:18)
      at HDSegwitBech32Wallet._getWIFByIndex (class/wallets/abstract-hd-electrum-wallet.js:93:23)
      at HDSegwitBech32Wallet._getExternalWIFByIndex (class/wallets/abstract-hd-electrum-wallet.js:77:19)
      at HDSegwitBech32Wallet._getWifForAddress (class/wallets/abstract-hd-wallet.ts:273:60)
      at PayjoinTransaction._callee$ (class/payjoin-transaction.js:27:38)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
      at node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7
      at tryCallTwo (node_modules/react-native/node_modules/promise/lib/core.js:45:5)
      at doResolve (node_modules/react-native/node_modules/promise/lib/core.js:200:13)
      at new Promise (node_modules/react-native/node_modules/promise/lib/core.js:66:3)
      at PayjoinTransaction.<anonymous> (node_modules/@babel/runtime/helpers/asyncToGenerator.js:21:12)
      at PayjoinTransaction.getPsbt (class/payjoin-transaction.js:33:131)
      at PayjoinClient.run (node_modules/payjoin-client/src/client.js:37:36)
      at _callee6$ (tests/unit/payjoin-transaction.test.js:123:29)
      at tryCatch (node_modules/regenerator-runtime/runtime.js:63:40)
      at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:294:22)
      at Generator.next (node_modules/regenerator-runtime/runtime.js:119:21)
      at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
      at node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7
      at tryCallTwo (node_modules/react-native/node_modules/promise/lib/core.js:45:5)
      at doResolve (node_modules/react-native/node_modules/promise/lib/core.js:200:13)
      at new Promise (node_modules/react-native/node_modules/promise/lib/core.js:66:3)
      at Object.<anonymous> (node_modules/@babel/runtime/helpers/asyncToGenerator.js:21:12)
      at TestScheduler.scheduleTests (node_modules/@jest/core/build/TestScheduler.js:333:13)
      at runJest (node_modules/@jest/core/build/runJest.js:404:19)
      at _run10000 (node_modules/@jest/core/build/cli/index.js:320:7)
      at runCLI (node_modules/@jest/core/build/cli/index.js:173:3)
```

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/pu3xg39fg668gqc0k9br3pf684?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9875)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/c2jk2faxafeamgpa7bcjgur6m0?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9876)

**Comment by limpbrains** [comment]
Do you think we can replace TRAVIS retry hack with `jest.retryTimes` ?

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/vz2qp1f6466yg1j20mta4yhbyg?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9877)

**Comment by Overtorment** [comment]
`detectOpenHandles` was barking at the wrong tree. actually, not ever once i remember `detectOpenHandles` to detect the issue correctly.
anyway, i debugged a bit and mocked the call the sets the timer

**Comment by Overtorment** [comment]
`jest.retryTimes` sounds like an awesome idea! didnt know such thing existed

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/rqvj59py7qh3mrmvy8mrxn7r7c?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9879)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/gythakqe5udnbbbwqvw4qy93g8?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9880)

**Comment by Overtorment** [comment]
welp, stupid circleci fails as well

**Comment by limpbrains** [comment]
Unfortunately, with `jest.retryTimes` jest does not retry if error is thrown in `beforeAll`. We need https://github.com/facebook/jest/issues/10520 to be implemented

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/nr1ajnaycczdg4qfrh0mvtamvw?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9888)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/cyve98z15qk98xe3ygup2u428r?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9891)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/gna68qwrdravz9r7uy7qpememr?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9895)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/k6htt81t6vjcgc0z19jg5u23m8?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9896)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/dk4qt4vdnubh9g4rmg0q98agrw?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9897)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/wradb4wqzfbp80t9ggkm6awcvm?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9898)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/3765z9ere3dkypzfjnndmrjme8?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9920)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/9ev1y7mgkbceaay6bk5n0wj9k0?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9921)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ty4dw7r8zy4qwpzymrdnb851w8?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9922)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/mexwavyzbgv2kk94nte6urjgdg?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9924)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/crc3hdpewy1b7kd0nhne52y6fc?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9925)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/98q8eu9c082yvzvm44xxc10num?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9937)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/z8w7p3btmtgpu6ta9r6gcjcu24?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9938)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/8yvrw8urbvfeuhj719unzjrhvr?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9939)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/mzh3pf9ht00gr19yqfr9fde14r?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9981)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/p9rqch8tmrdem34vh9zavc2tf0?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9991)

**Comment by limpbrains** [comment]
For some reason tests became super slow on circleci
before: 
```
 PASS  tests/unit/hd-legacy-wallet.test.js (172.796 s)
 PASS  tests/unit/deeplink-schema-match.test.js (173.208 s)
 PASS  tests/unit/storage.test.js (173.395 s)
 ```
 after:
 ```
 PASS  tests/unit/hd-legacy-wallet.test.js (805.092 s)
 PASS  tests/unit/deeplink-schema-match.test.js (803.999 s)
 PASS  tests/unit/storage.test.js (801.805 s)
```

That is why circleci fails with timeout error.
I thought this is because of nodejs bug https://github.com/facebook/jest/issues/11956, but downgrading node to 16.10 doesn't help

github actions is fine thought

**Comment by limpbrains** [comment]
No idea why e2e test are falling. Looks like detox successfully installs app to the emulator, but can't launch it
```
18:19:24.714 detox[9037] TRACE: [EXEC_SUCCESS, #17] /Users/runner/work/BlueWallet/BlueWallet/android/app/build/outputs/apk/release/app-release.apk: 1 file pushed, 0 skipped. 21.9 MB/s (76886074 bytes in 3.342s)

18:22:25.901 detox[9037] ERROR: ChildProcessError: `/Users/runner/Library/Android/sdk/platform-tools/adb -s emulator-5554 shell pm install -r -g -t /data/local/tmp/detox/Application.apk` failed with code null
    at ChildProcess.<anonymous> (/Users/runner/work/BlueWallet/BlueWallet/node_modules/child-process-promise/lib/index.js:132:23)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Socket.<anonymous> (node:internal/child_process:451:11)
    at Socket.emit (node:events:527:28)
    at Pipe.<anonymous> (node:net:709:12)
```

File: tests/unit/loc.test.js [hunk 1]
```diff
@@ -1,9 +1,8 @@
+import assert from 'assert';
 import { BitcoinUnit } from '../../models/bitcoinUnits';
 import { FiatUnit } from '../../models/fiatUnit';
 import { _leaveNumbersAndDots, formatBalanceWithoutSuffix, formatBalancePlain, formatBalance } from '../../loc';
-const assert = require('assert');
 const currency = require('../../blue_modules/currency');
-jest.useFakeTimers();
```
Comment (Overtorment) [inline tests/unit/loc.test.js:6]: hmm. i dont remember, do we have timers on loc..?

File: tests/integration/hd-segwit-bech32-wallet.test.js [hunk 2]
```diff
@@ -251,7 +251,7 @@ describe('Bech32 Segwit HD (BIP84)', () => {
     assert.strictEqual(outputs[outputs.length - 1].address, changeAddress);
   });
 
-  it('wasEverUsed() works', async () => {
+  it.only('wasEverUsed() works', async () => {
```
Comment (Overtorment) [inline tests/integration/hd-segwit-bech32-wallet.test.js:254]: oops, test code

File: tests/e2e/bluewallet.spec.js [hunk 3]
```diff
@@ -12,17 +12,11 @@ import {
...
 
-jasmine.getEnv().addReporter({
-  specStarted: result => (jasmine.currentTest = result),
-  specDone: result => (jasmine.currentTest = result),
-});
-
 describe('BlueWallet UI Tests - no wallets', () => {
   it('selftest passes', async () => {
-    const lockFile = '/tmp/travislock.' + hashIt(jasmine.currentTest.fullName);
+    const lockFile = '/tmp/travislock.' + hashIt('t1');
```
Comment (Overtorment) [inline tests/e2e/bluewallet.spec.js:17]: eh, this will have to do

File: .circleci/config.yml [hunk 4]
```diff
@@ -20,4 +20,6 @@ jobs:
             - node_modules
 
       # run tests!
-      - run: npm t || npm t || npm t
+      - run:
+          command: npm t || npm t || npm t
+          no_output_timeout: 60m
```
Comment (Overtorment) [inline .circleci/config.yml:25]: maybe circleci is slow because its throttling freeloaders like us

**Review by Overtorment — APPROVED** [review]

**Comment by Overtorment** [comment]
im seeing discrepancies: `app-release.apk` and `Application.apk` 
could it be it..?

ill try to work on this, at least will try to run in on my localhost

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/165eypcat365kgg5v6ndy6ekpw?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9993)

**Comment by limpbrains** [comment]
For detox tests we can use `detox --retries N` https://wix.github.io/Detox/docs/api/detox-cli/#test
It doesn't perfectly fits here, because it re-rust ALL tests in file if one of the fails.
```
This is not a recommended method since it assumes, for example, that if in a 100-tests suite even 1 test fails, we will be re-running the entire test file with those 100 tests again.
```

But if we split .spec.js files into many, each file one separate test, we can remove these hacks.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/gprmhxbdh6c2y42rh28kzuv4xw?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/10000)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ujap9r1ua5qdzuv5qr6618qq68?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/10002)

**Comment by Overtorment** [comment]
getting weird  `2022-05-24 21:33:14.443 qemu-system-x86_64-headless[11606:79520] *** Terminating app due to uncaught exception 'NSRangeException', reason: '*** -[__NSArray0 objectAtIndex:]: index 0 beyond bounds for empty NSArray'`

works ok on my localhost

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ja1qkpq9fchxe15v1ptq2u5jpw?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/10003)

**Comment by Overtorment** [comment]
gona do some experiments here : https://github.com/BlueWallet/BlueWallet/pull/4754

**Comment by Overtorment** [comment]
got some luck in https://github.com/BlueWallet/BlueWallet/pull/4754
I managed to make it work, I re-integrated detox to android from scratch.

**Comment by limpbrains** [comment]
Awesome. Please note that I downgraded detox in last commit, probably you can try to revert it

**Comment by Overtorment** [comment]
merged in https://github.com/BlueWallet/BlueWallet/pull/4754

**Pull request closed without merging**